specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/azureFirewall.json

```diff
@@ -473,6 +473,128 @@
          },
          "description":"Properties of an application rule."
       },
+      "AzureFirewallNatRuleCollectionPropertiesFormat": {
+         "properties":{  
+            "priority":{  
+               "type":"integer",
+               "format":"int32",
+               "maximum":65000,
+               "exclusiveMaximum":false,
+               "minimum":100,
+               "exclusiveMinimum":false,
+               "description":"Priority of the NAT rule collection resource."
+            },
+            "action":{  
+               "$ref":"#/definitions/AzureFirewallNatRCAction",
+               "description":"The action type of a NAT rule collection"
+            },
+            "rules":{  
+               "type":"array",
+               "items":{  
+                  "$ref":"#/definitions/AzureFirewallNatRule"
+               },
+               "description":"Collection of rules used by a NAT rule collection."
+            },
+            "provisioningState":{  
+               "description":"The provisioning state of the resource.",
+               "$ref":"#/definitions/ProvisioningState"
+            }
+         },
+         "description":"Properties of the NAT rule collection."
+      },
+      "AzureFirewallNatRuleCollection":{  
+         "properties":{  
+            "properties":{  
+               "x-ms-client-flatten":true,
+               "$ref":"#/definitions/AzureFirewallNatRuleCollectionPropertiesFormat"
+            },
+            "name":{  
+               "type":"string",
+               "description":"Gets name of the resource that is unique within a resource group. This name can be used to access the resource."
+            },
+            "etag":{  
+               "type":"string",
+               "readOnly":true,
+               "description":"Gets a unique read-only string that changes whenever the resource is updated."
+            }
+         },
+         "allOf":[  
+            {  
+               "$ref":"./network.json#/definitions/SubResource"
+            }
+         ],
+         "description":"NAT rule collection resource"
+      },
+      "AzureFirewallNatRule":{  
+         "properties":{  
+            "name":{  
+               "type":"string",
+               "description":"Name of the NAT rule."
+            },
+            "description":{  
+               "type":"string",
+               "description":"Description of the rule."
+            },
+            "sourceAddresses":{  
+               "type":"array",
+               "description":"List of source IP addresses for this rule.",
+               "items":{  
+                  "type":"string"
+               }
+            },
+            "destinationAddresses":{  
+               "type":"array",
+               "description":"List of destination IP addresses for this rule.",
+               "items":{  
+                  "type":"string"
+               }
+            },
+            "destinationPorts":{  
```

Comment:
nit: ports generally have a certain format, if this property is not readonly, it's a good idea to define a regex pattern for it
This applies to all other port/address types